core/src/impl/Kokkos_Error.hpp

```diff
@@ -55,6 +55,12 @@
 #define KOKKOS_ABORT_MESSAGE_BUFFER_SIZE 2048
 #endif  // ifndef KOKKOS_ABORT_MESSAGE_BUFFER_SIZE
 
+// HIP defines a macro abort() when using nvcc, thus we get a collision with the
+// abort function defined here
+#ifdef KOKKOS_ENABLE_HIP
```

Comment:
OMG Did you file a bug report on this? I mean how in the world can you think defining a macro called "abort" is an acceptable idea???
Add some comment like // KOKKOS_IMPL_HIP_WORKAROUND

Reply:
No, I haven't. For reference here is the [guilty code](https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/nvcc_detail/hip_runtime.h#L104-L114).

Reply:
wow. Just wow. They could at least make it a freaking function so you only get shadowing warnings instead of compile errors ...